code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -844,6 +845,343 @@
 
 	return ..()
 
+/obj/effect/alien/resin/king_cocoon
+	name = "alien cocoon"
+	desc = "A large pulsating cocoon."
+	icon = 'icons/obj/structures/alien/xenoKingHatchery.dmi'
+	icon_state = "static"
+	health = 4000
+	pixel_x = -48
+	pixel_y = -64
+	density = TRUE
+	plane = FLOOR_PLANE
+
+	/// The mob picked as a candidate to be the King
+	var/client/chosen_candidate
+	/// The hive associated with this cocoon
+	var/hive_number = XENO_HIVE_NORMAL
+	/// Whether the cocoon has hatched
+	var/hatched = FALSE
+	/// Current running timer
+	var/timer
+	/// Is currently rolling candidates
+	var/rolling_candidates = FALSE
+	/// Voting for King
+	var/list/mob/living/carbon/xenomorph/votes = list()
+
+/obj/effect/alien/resin/king_cocoon/Destroy()
+	if(!hatched)
+		marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP IN [get_area_name(loc)] HAS BEEN STOPPED.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
```

Comment:
```suggestion
		marine_announcement("ALERT.\n\nUNSUAL ENERGY BUILDUP IN [uppertext(get_area_name(loc))] HAS BEEN STOPPED.", "[MAIN_AI_SYSTEM] Biological Scanner", 'sound/misc/notice1.ogg')
```
they sure look silly ingame though, not sure why you wouldnt want to capitalize em